chain/epoch_manager/src/proposals.rs

```diff
@@ -72,18 +72,25 @@ pub fn proposals_to_epoch_info(
     }
 
     // Get the threshold given current number of seats and stakes.
-    let num_fisherman_seats: u64 = epoch_config.avg_fisherman_per_shard.iter().sum();
-    let num_seats = epoch_config.num_block_producers + num_fisherman_seats;
+    let num_hidden_validator_seats: u64 = epoch_config.avg_hidden_validators_per_shard.iter().sum();
+    let num_seats = epoch_config.num_block_producers + num_hidden_validator_seats;
     let stakes = ordered_proposals.iter().map(|(_, p)| p.amount).collect::<Vec<_>>();
     let threshold = find_threshold(&stakes, num_seats)?;
     // Remove proposals under threshold.
     let mut final_proposals = vec![];
+    let mut fishermen = vec![];
+    let mut fishermen_to_index = HashMap::new();
     for (account_id, p) in ordered_proposals {
         if p.amount >= threshold {
             if !stake_change.contains_key(&account_id) {
                 stake_change.insert(account_id, (p.amount, 0));
             }
             final_proposals.push(p);
+        } else if p.amount >= epoch_config.fishermen_threshold {
+            // Do not return stake back since they will become fishermen
+            stake_change.entry(account_id.clone()).or_insert((p.amount, 0));
+            fishermen_to_index.insert(p.account_id.clone(), fishermen.len() as ValidatorId);
```

Comment:
Why `account_id` in the first line and `p.account_id` in the 2nd?

Reply:
Because I am a n00b :)